dev-tools-java/src/test/java/org/jboss/forge/addon/devtools/java/JavaFormatSourcesCommandTest.java

```diff
@@ -0,0 +1,196 @@
+import org.jboss.forge.furnace.repositories.AddonDependencyEntry;
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.FileAsset;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+public class JavaFormatSourcesCommandTest
+{
+   @Deployment
+   @Dependencies({
+            @AddonDependency(name = "org.jboss.forge.addon:parser-java"),
+            @AddonDependency(name = "org.jboss.forge.addon:dev-tools-java"),
+            @AddonDependency(name = "org.jboss.forge.addon:ui-test-harness"),
+            @AddonDependency(name = "org.jboss.forge.addon:projects"),
+            @AddonDependency(name = "org.jboss.forge.addon:maven"),
+            @AddonDependency(name = "org.jboss.forge.furnace.container:cdi"),
+            @AddonDependency(name = "org.jboss.forge.addon:shell-test-harness")
+   })
+   public static ForgeArchive getDeployment()
+   {
+
+      ForgeArchive archive = ShrinkWrap
+               .create(ForgeArchive.class)
+               .addBeansXML()
+               .add(new FileAsset(new File(
+                        "src/test/resources/org/jboss/forge/addon/devtools/java/resources/FormattedSource.java")),
+                        "org/jboss/forge/addon/devtools/java/FormattedSource.java")
+               .add(new FileAsset(new File(
+                        "src/test/resources/org/jboss/forge/addon/devtools/java/resources/UnformattedSource.java")),
+                        "org/jboss/forge/addon/devtools/java/UnformattedSource.java")
+               .add(new FileAsset(
+                        new File(
+                                 "src/test/resources/org/jboss/forge/addon/devtools/java/resources/DefaultFormattedSource.java")),
+                        "org/jboss/forge/addon/devtools/java/DefaultFormattedSource.java")
+               .add(new FileAsset(new File(
+                        "src/test/resources/org/jboss/forge/addon/devtools/java/resources/FormatProfile.xml")),
+                        "org/jboss/forge/addon/devtools/java/FormatProfile.xml")
+               .addAsAddonDependencies(
+                        AddonDependencyEntry.create("org.jboss.forge.furnace.container:cdi"),
+                        AddonDependencyEntry.create("org.jboss.forge.addon:dev-tools-java"),
+                        AddonDependencyEntry.create("org.jboss.forge.addon:maven"),
+                        AddonDependencyEntry.create("org.jboss.forge.addon:ui-test-harness"),
+                        AddonDependencyEntry.create("org.jboss.forge.addon:shell-test-harness"),
+                        AddonDependencyEntry.create("org.jboss.forge.addon:resources")
+               );
+      return archive;
+   }
+
+   @Inject
+   private ResourceFactory resourceFactory;
+
+   @Inject
+   private UITestHarness testHarness;
+
+   @Inject
+   private ShellTest shellTest;
```

Comment:
@gastaldi  I have used Shell Test in the test:

public void testFileFormattingWithoutProfileAndSourceInput() throws Exception (L185).

Thats why I added the shell-test-harness dependency too.